src/js/player.js

```diff
@@ -1092,6 +1092,8 @@ class Player extends Component {
       this.addClass('vjs-fullscreen');
     } else {
       this.removeClass('vjs-fullscreen');
+      // Making sure the Control text reverts back after pressing ESC
+      this.controlBar.fullscreenToggle.controlText('Fullscreen');
```

Comment:
the fullscreen toggle [handles](https://github.com/videojs/video.js/blob/fc72fb90e7661594c0ea090e0e1e9838f8ec9486/src/js/control-bar/fullscreen-toggle.js#L30-L38) the control text changes. We should make that component handle this use case correctly as well.